### ISSUE TITLE
.Net Core Support

### ISSUE THREAD

**Issue by aabenoja** [SparkViewEngine/spark#60]
I have a side project using ASP .Net Core, and I'm looking to use spark rather than razor for my custom middleware stack. I haven't started looking into the steps to start porting this over, so I don't know how involved the process would be, or if there would be enough support for this.

**Comment by RobertTheGrey** [comment]
I'm currently waiting for dotnetcore to settle down for I do anything with this. If you are happy to do the work then that's great, but with all the changes recently, I'm expecting there may be more before the RC goes RTW

**Comment by aabenoja** [comment]
I thought it had already settled down with [the 1.0.0 announcement](https://blogs.msdn.microsoft.com/dotnet/2016/06/27/announcing-net-core-1-0/)?

**Comment by RobertTheGrey** [comment]
Yes, you're right the core is at v1 but the tooling is still in preview and a bunch of the supporting Nugets are still not at v1. I'm concerned I spent time on it just to have to change things again, but if you're in the mood to check it out, take a look at [this](https://icanhasdot.net/faq) and let me know your thoughts. If you're keen to spend time on it, I won't stop you :+1:

**Comment by aabenoja** [comment]
Cool, didn't notice the tooling was still in preview. Good to know. I'll take a stab at it, see where I get.

**Comment by RainingNight** [comment]
Any new message?

**Comment by RobertTheGrey** [comment]
Currently working on a port of the code to .Net Core 2. Won't be supporting original .Net Core, and I'd say the port is currently about 50% done removing dependencies on System.Web being the tricky part.

**Comment by bounav** [comment]
@RobertTheGrey how are things going on the .net core front?

Anything the community could do to help?

**Comment by RobertTheGrey** [comment]
Was going strong in Feb, but March was full of holidays, so I'll be getting back into it again now in April - feel like it's about 70% there, so hopefully have something out to test in coming weeks

**Comment by bounav** [comment]
@RobertTheGrey how are things progressing? 

We're eager to port some of our website to .net core and spark view engine is the last dependency holding us back!
Going back to razor view engine would be such a sad thing to do ;).

I'll re-iterate my offer to help: Are there any bits the community could do to assist?

**Comment by RobertTheGrey** [comment]
Hey @bounav - thanks for the nudge! I completely stalled on this due to work pressure at the back end of last year. I still have a couple of work projects on the go now that are time-sensitive for the next month or two, so it's tricky to know when I'm going to get time to sit down and look at this.

Plus with the whole Blazor thing, I didn't really think anyone was waiting on this to be fair, so I haven't been pushing myself to get through it. But it does seem a shame to leave it incomplete, especially when it can benefit production systems.

If you drop me an email, I can send you my mobile (UK) and we can talk about the features you're using in your code base as some were more difficult to port than others when I was in the thick of it, and we can see what the best approach may be. If there's something you can do to get it production ready, then we'll do that, and at least you're not stuck waiting for me to get some free time.

Cheers!

**Comment by bounav** [comment]
Really nice chatting with you over the phone!

As discussed we use the following Spark features:

- In ASP.NET 5 (at the moment) we configure `SparkViewEngine` so that we don't have to write Razor (.cshtml) views
- Our `.spark` views make use of

    - Partial views
    - 3 pass rendering

The list above represents, at least for to me and my team, the minimum feature set for a first attempt at getting spark to work in .NET core.

We also talked about the more advanced spark features (E.g. bindings and the use of other programming languages)... but this should probably be ported/extended at a later point?

One of the thing's I'll try to do in the coming weeks is to generate a small ASP.NET Core test project that includes a couple of Razor page that mimics a "real" use case. 
It's going to be just a few views, nothing else but that should help testing and getting the ball rolling.

**Comment by RobertTheGrey** [comment]
Thanks @bounav - yes great chat thanks. I'll look out for the stuff we talked through and if I make some progress I'll be sure to post it here.

**Comment by mikernet** [comment]
For what it's worth, this was a great view engine with tidy syntax that I much preferred over Razor. A bit sad it hasn't been updated so I can't use it but if a new version ever gets released that works with .NET Core / .NET 5, I'd love to start using it again :)

**Comment by neilrees** [comment]
Can I ask what the current state of this repository is?  @bounav Has done a load of good work to update the project and add .NET Core targets - is it in a position where a new release could be made?

**Comment by bounav** [comment]
@neilrees watch PR #79 !

**Comment by neilrees** [comment]
With https://github.com/SparkViewEngine/spark/pull/79 having been merged 🎉 , is there likely to be a new release?

**Comment by bounav** [comment]
Yes, I need to find the time to work on the deployment pipeline. Once sorted out we're thinking of release a new pre-release version of spark, both for MVC targeting _.net framework_ and _aspnetcore_.

**Comment by neilrees** [comment]
@bounav Is there anything we can do help put a build together?
